### PR TITLE
Javascript javascript1 week2

### PR DESCRIPTION
**Pull request by tonevrav**
Added homework files:

- full-name.js
- event-app.js
- weather-wear.js
- weather-wear-v2.js
- student-manager.js
- candy-helper.js

### CODE REVIEW

File: javascript/javascript1/week2/homework/candy-helper.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        case "Toffee":
+            boughtCandyPrices.push(1.1 * weight);
+        break;
+        case "Chewing-gum":
+            boughtCandyPrices.push(0.03 * weight);
+        break;
+        default:
+            console.log("Sorry, something went wrong :(");
+    }
+}
```
Comment (Rassibassi) [inline javascript/javascript1/week2/homework/candy-helper.js:21]: Good job, you could lower case the `candyType` just before the `switch` statement, then things are not case sensitive.

File: javascript/javascript1/week2/homework/candy-helper.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        total += price;
+    }
+
+    if(total > amountToSpend) {
+        console.log("Enough candy for you!");
+        return false;
+    } else {
+        console.log("You can buy more, so please do!");
+        return true;
+    }
```
Comment (Rassibassi) [inline javascript/javascript1/week2/homework/candy-helper.js:36]: If an `else` statement is the last thing in a function and entails a `return`, you can actually just have an `if` and then in the function scope whatever is in the `else` without the `else` scope

File: javascript/javascript1/week2/homework/event-app.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    let indexOfEventDay = currentDay + remainingDays; // 6 + 1 = 7
+
+    if(indexOfEventDay >= 7) { // true
+    indexOfEventDay -= 7; // 7 - 7 = 0
+    }
+
+    return days[indexOfEventDay]; // days[0] (Sunday)
+}
+
```
Comment (Rassibassi) [inline javascript/javascript1/week2/homework/event-app.js:16]: line 10-12 is not needed if you do the modulus after the addition

File: javascript/javascript1/week2/homework/full-name.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            return `Lord ${fullName}`;
+        } else if (gender === "female") {
+            return `Lady ${fullName}`;
+        } else {
+            return "Please, indicate your gender.";
+        }
+    }
+
+    return fullName;
+}
```
Comment (Rassibassi) [inline javascript/javascript1/week2/homework/full-name.js:20]: `useFormalName === true` is the same as `useFormalName`

File: javascript/javascript1/week2/homework/student-manager.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            console.log("Cannot add more students to class 07");
+        } else {
+            if(studentName === "") {
+                console.log("You cannot add an empty string to a class");
+            } else {
+                class07Students.push(studentName);
+            }
+        }
+    }
+}
```
Comment (Rassibassi) [inline javascript/javascript1/week2/homework/student-manager.js:20]: You can make it easier for yourselves, if you check for all the negatives first, and then, if no negatives are left do the push. There are too many nested `if` s

File: javascript/javascript1/week2/homework/weather-wear.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    } else if(temperature < 15) {
+        return "Light jacket";
+    } else if(temperature < 25) {
+        return "Shirt and shorts";
+    } else if(temperature >= 25) {
+        return "Sunglases";
+    } else{
+        return "Ooops...";
+    }
+}
```
Comment (Rassibassi) [inline javascript/javascript1/week2/homework/weather-wear.js:17]: Good job, can you think of a nested array with strings and temperatures side by side, and then a for loop with a single `if` inside which does the trick?

**Review by Rassibassi — COMMENTED** [review]
Good job, see my inline comments

File: javascript/javascript1/week2/homework/weather-wear.js (hunk 6 above)
Comment (tonevrav) [reply javascript/javascript1/week2/homework/weather-wear.js:17]: Thank you Rasmus for your review and comments! And please, check below to see if I've solved your additional task:

`function whatToPutOn(temperature) {
    const clothingTemp = [
        ["extra-warm jumpsuit", -20],
        ["warm jumpsuit", -10],
        ["warm trousers", 5],
        ["light jacket", 15],
        ["shirt and shorts", 25],
        ["sunglasses", Infinity]
    ];

    for(let i = 0; i < clothingTemp.length; i+=1) {
        if(temperature <= clothingTemp[i][1]){
            return `Put ${clothingTemp[i][0]} on!`;
        }
    }
}`

**Pull request merged**